### PR TITLE
334 rle

### PR DESCRIPTION
**Pull request by alanbernstein**

### CODE REVIEW

File: roaring/roaring.go [hunk 1]
```diff
@@ -1267,12 +1267,33 @@ func (c *container) bitmapToRun() {
...
+		if v - c.array[i-1] > 1 {
+			// if current-previous > 1, one run ends and another begins
+			c.runs = append(c.runs, interval32{start, c.array[i-1]})
+			start = v
+		}
+	}
+	// append final run
+	c.runs = append(c.runs, interval32{start, c.array[i]})
+	c.array = nil
+	c.mapped = false
```
Comment (alanbernstein) [inline roaring/roaring.go]: I'm not exactly sure about the implications of this, I copied it from the existing conversion functions

Comment (jaffee) [inline roaring/roaring.go]: probably a good call

Comment (tgruben) [inline roaring/roaring.go]: i'm surprised that works, seems to me like the first iteration of that loop would throw an index out of bounds exception

File: roaring/roaring.go [hunk 2]
```diff
@@ -1267,12 +1267,33 @@ func (c *container) bitmapToRun() {
...
+	start := c.array[0]
+	for i, v := range c.array[1:] {
+		if v - c.array[i] > 1 {
+			// if current-previous > 1, one run ends and another begins
+			c.runs = append(c.runs, interval32{start, c.array[i]})
+			start = v
+		}
+	}
+	// append final run
+	c.runs = append(c.runs, interval32{start, c.array[c.n-1]})
```
Comment (tgruben) [inline roaring/roaring.go]: I would prefer c.array[len(c.array)-1].  Though I guess we should trust c.n to be accurate.

Comment (alanbernstein) [inline roaring/roaring.go]: yea, if we can't trust c.n, we should address that rather than avoid it here

**Pull request merged**